### PR TITLE
Allow allow downloading from a list of possible URLs

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-client.c [hunk 1]
```diff
@@ -2473,7 +2476,7 @@ fwupd_client_install_release_download_cb (GObject *source, GAsyncResult *res, gp
 }
 
 static gboolean
-fwupd_client_is_url (const gchar *perhaps_url)
+fwupd_client_is_url_http (const gchar *perhaps_url)
```
Comment (superm1) [inline libfwupd/fwupd-client.c:2479]: doesn't this function also handle directories?

Comment (hughsie) [reply libfwupd/fwupd-client.c:2479]: I don't think so; `fwupd_client_install_release_remote_cb` uses it as `fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_LOCAL && !fwupd_client_is_url_http (fwupd_release_get_uri (data->release)` i.e. a directory would return `FALSE` from `fwupd_client_is_url()`

File: libfwupd/fwupd-client.c [hunk 2]
```diff
@@ -4228,56 +4231,131 @@ fwupd_client_download_write_callback_cb (char *ptr, size_t size, size_t nmemb, v
...
 	g_task_return_pointer (task,
-			       g_byte_array_free_to_bytes (g_steal_pointer (&buf)),
+			       g_steal_pointer (&blob),
 			       (GDestroyNotify) g_bytes_unref);
 }
 #endif
 
+/* private */
+void
+fwupd_client_download_bytes2_async (FwupdClient *self,
```
Comment (superm1) [inline libfwupd/fwupd-client.c:4319]: presumably this should change with the ABI bump and not be the private one anymore.

**Review by superm1 — APPROVED** [review]

File: libfwupd/fwupd-client.c (hunk 2 above)
Comment (hughsie) [reply libfwupd/fwupd-client.c:4319]: Yes, probably; let's see if we need any more parameters before we export it.

**Pull request merged**